lib/shared/addon/components/container/form-scheduling/component.js

```diff
@@ -27,96 +19,74 @@ export default Component.extend(ManageLabels, {
   canRequestHost: true,
 
-  // Initial labels and host to start with
-  initialLabels: null,
+  // Initial host to start with
   initialHostId: null,
 
-  // labelArray -> the labels that should be set for the scheduling rules
-
-  // Actions output
-  // setLabels(labelArray)
-  // setRequestedHost(hostId)
-
   // Internal properties
   isRequestedHost: false,
 
   editing: true,
 
-  classNames: ['accordion-wrapper'],
-
-  actions: {
-    addSchedulingRule() {
-      this.send('addAffinityLabel');
-    },
+  advanced: false,
 
-    removeSchedulingRule(obj) {
-      this.send('removeLabel', obj);
-    },
-  },
+  classNames: ['accordion-wrapper'],
 
   init() {
-    this.set('allHosts', this.get('store').all('host'));
     this._super(...arguments);
-
-    this.initLabels(this.get('initialLabels'), 'affinity');
-
-    if ( this.get('isGlobal') )
-    {
-      this.setProperties({
-        isRequestedHost: false,
-        requestedHostId: null,
-      });
-      scheduleOnce('afterRender', () => {
-        this.sendAction('setRequestedHost', null);
-      });
+    this.loadHosts();
+    this.set('advanced', !this.get('editing'));
+    if (this.get('initialHostId')) {
+      debugger
+      this.set('requestedHostId', this.get('initialHostId'));
     }
-    else if ( this.get('initialHostId') )
-    {
-      this.setProperties({
-        isRequestedHost: true,
-        requestedHostId: this.get('initialHostId'),
-      });
+    if (!this.get('scheduling.node')) {
+      this.set('scheduling.node', {});
     }
   },
 
   didReceiveAttrs() {
     if (!this.get('expandFn')) {
-      this.set('expandFn', function(item) {
-          item.toggleProperty('expanded');
+      this.set('expandFn', function (item) {
+        item.toggleProperty('expanded');
       });
     }
   },
 
-  updateLabels(labels) {
-    this.sendAction('setLabels', labels);
+  loadHosts() {
+    const clusterId = this.get('scope.currentCluster.id');
+    this.get('globalStore').findAll('machine', { filter: { clusterId: this.get('scope.currentCluster.id') } }).then((hosts) => {
+      this.set('allHosts', hosts);
+    });
   },
 
-  isRequestedHostDidChange: function() {
-    if ( this.get('isRequestedHost') )
-    {
+  isRequestedHostDidChange: function () {
+    const scheduling = this.get('scheduling');
+    if (this.get('isRequestedHost')) {
       var hostId = this.get('requestedHostId') || this.get('hostChoices.firstObject.id');
+      Object.keys(scheduling).forEach(key => {
+        delete scheduling[key];
+      });
       this.set('requestedHostId', hostId);
     }
-    else
-    {
+    else {
       this.set('requestedHostId', null);
+      delete scheduling['nodeId'];
     }
   }.observes('isRequestedHost'),
 
-  requestedHostIdDidChange: function() {
+  requestedHostIdDidChange: function () {
     var hostId = this.get('requestedHostId');
-    this.sendAction('setRequestedHost', hostId);
+    this.set('scheduling.nodeId', hostId);
```

Comment:
`scheduling.node.nodeId` (and elsewhere, the workload.nodeId  is/will be readonly, workload.scheduling.node.nodeId is where you request one)